src/boot/efi/addon.c

```diff
@@ -0,0 +1,10 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "efi.h"
+
+/* This is intended to carry data, not to be executed */
+
+EFIAPI EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *system_table);
+EFIAPI EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *system_table) {
+    return EFI_SECURITY_VIOLATION;
```

Comment:
Returning a security violation seems a bit odd to me. `EFI_ABORTED` or `EFI_UNSUPPORTED` sounds more appropriate.